### ISSUE TITLE
Segmentation fault in MMFF

### ISSUE THREAD

**Issue by Meteor-han** [rdkit/rdkit#6728]
**Describe the bug**
After embedding a molecule to get its 3D coordinates, MMFF optimization gets "Segmentation fault (core dumped)".

**To Reproduce**
>>> from rdkit import Chem
>>> from rdkit.Chem import AllChem
>>> smiles = "CC(C)(C)CO[H-]F"
>>> mol_ = Chem.MolFromSmiles(smiles)
>>> mol = Chem.AddHs(mol_)
>>> allconformers = AllChem.EmbedMultipleConfs(mol, numConfs=1)
>>> AllChem.MMFFOptimizeMolecule(mol, confId=0)
Segmentation fault (core dumped)

**Expected behavior**
No segmentation fault. Hence we can use try-except to capture errors if AllChem.MMFFOptimizeMolecule fails.

**Configuration:**
 - RDKit version: 2023.3.3
 - OS: Ubuntu 20.04
 - Python version: 3.10
 - Are you using conda? Yes, installed rdkit with pip install rdkit

**Comment by bp-kelley** [comment]
It seg faults on osx as well.

**Comment by bp-kelley** [comment]
It's crashing in the atom typer

https://github.com/rdkit/rdkit/blob/ccf450e236e87fd3cc4a31adaee10fdc372c31bb/Code/GraphMol/ForceFieldHelpers/MMFF/AtomTyper.cpp#L3778

The strechbend lookup is failing and returning a null pointer which is then dereferenced.

**Comment by bp-kelley** [comment]
I believe this code should be:
```
             if(mmffStbnParams.second) {
                  res = (!(isDoubleZero((mmffStbnParams.second)->kbaIJK) &&
                           isDoubleZero((mmffStbnParams.second)->kbaKJI)));
              } else {
                  res = false;
              }
```

**Comment by Meteor-han** [comment]
@bp-kelley Thank you for the prompt reply. Currently, I am abandoning the use of MMFF until the bug is fixed.

**Comment by ptosco** [comment]
I will fix this bug before 2023.09 release. Thanks for reporting!
